judge/models/profile.py

```diff
@@ -235,14 +235,14 @@ def calculate_points(self, table=_pp_table):
                            .annotate(max_points=Max('submission__points')).order_by('-max_points')
                            .values_list('max_points', flat=True).filter(max_points__gt=0)
         )
-        extradata = (
-            public_problems.filter(submission__user=self, submission__result='AC').values('id').distinct().count()
-        )
         bonus_function = settings.DMOJ_PP_BONUS_FUNCTION
         points = sum(data)
-        problems = len(data)
         entries = min(len(data), len(table))
-        pp = sum(map(mul, table[:entries], data[:entries])) + bonus_function(extradata)
+        problems = (
+            public_problems.filter(submission__user=self, submission__result='AC', submission__points__gte=F('points'))
```

Comment:
Can we base this off case points instead? (Yes, I am aware we use `submission__points` elsewhere, but we should change that too)

Reply:
Ok, I changed it to `submission__case_points__gte=F('submission__case_total')`.